atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/TimestampsByCellResultWithToken.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.keyvalue.dbkvs.impl;
+
+import java.util.Arrays;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Iterators;
+import com.google.common.collect.PeekingIterator;
+import com.google.common.collect.SetMultimap;
+import com.google.common.primitives.UnsignedBytes;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.common.base.ClosableIterator;
+import com.palantir.nexus.db.sql.AgnosticLightResultRow;
+
+final class TimestampsByCellResultWithToken {
+    private static final String ROW = "row_name";
+    private static final String COL = "col_name";
+    private static final String TIMESTAMP = "ts";
+
+    private byte[] currentRow = null;
+    private byte[] currentCol = null;
+    private Long currentTimestamp = null;
+    private PeekingIterator<AgnosticLightResultRow> iterator;
+
+    final SetMultimap<Cell, Long> entries;
+    boolean mayHaveMoreResults = false;
```

Comment:
Should probably expose getters for these and make them private, to avoid modification.